alf/utils/common.py

```diff
@@ -433,6 +433,31 @@ def _cast(obs):
     return tf.nest.map_structure(_cast, observation)
 
 
+def traverse_transform_observation(obs, levels, field, func):
+    """
+    Args:
+        obs (nested Tensor|nested ndarray): observations to be applied the transformation
+        levels (list[str]): levels
+        field (str): full field path
+        func (Callable): transform func
+    """
+    if not levels:
+        return func(obs, field)
+    level = levels[0]
+    if isinstance(obs, tuple) and hasattr(obs, '_fields'):
```

Comment:
Perhaps add a utility function `is_namedtuple()`?

Reply:
added `is_namedtuple` in `nest_utils.py`